servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectLBHttpConnectionFactory.java

```diff
@@ -0,0 +1,174 @@
+                    processor = newSingleProcessor();
+            final Channel channel = connection.nettyChannel();
+            assert channel.eventLoop().inEventLoop();
+            channel.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                @Override
+                public void userEventTriggered(final ChannelHandlerContext ctx, final Object evt) {
+                    if (evt instanceof SslHandshakeCompletionEvent) {
+                        SslHandshakeCompletionEvent event = (SslHandshakeCompletionEvent) evt;
+                        if (event.isSuccess()) {
+                            processor.onSuccess(connection);
+                        } else {
+                            processor.onError(event.cause());
+                        }
+                        channel.pipeline().remove(this);
+                    }
+                    ctx.fireUserEventTriggered(evt);
+                }
+            });
+
+            final Single<FilterableStreamingHttpConnection> result;
+            final DeferSslHandler deferSslHandler = channel.pipeline().get(DeferSslHandler.class);
+            if (deferSslHandler == null) {
+                if (!channel.isActive()) {
+                    result = Single.failed(StacklessClosedChannelException.newInstance(connection +
+                                    " Connection is closed, either received a 'Connection: closed' header or" +
+                                    " closed by the proxy. Investigate logs on a proxy side to identify the cause.",
+                            ProxyConnectLBHttpConnectionFactory.class, "handshake"));
+                } else {
+                    result = Single.failed(new IllegalStateException(connection +
+                            " Unexpected connection state: failed to find a handler of type " +
+                            DeferSslHandler.class + " in the channel pipeline."));
+                }
+            } else {
+                deferSslHandler.ready();
+                result = fromSource(processor);
+            }
+            return result.shareContextOnSubscribe();
+        });
+    }
+
+    private static Single<FilterableStreamingHttpConnection> drainPropagateError(
+            final StreamingHttpResponse response, final Throwable error) {
+        return safeCompletePropagateError(response.messageBody().ignoreElements(), error);
+    }
+
+    private static Single<FilterableStreamingHttpConnection> closePropagateError(
+            final FilterableStreamingHttpConnection connection, final Throwable error) {
+        return safeCompletePropagateError(connection.closeAsync(), error);
+    }
+
+    private static Single<FilterableStreamingHttpConnection> safeCompletePropagateError(
+            final Completable completable, final Throwable error) {
+        return completable
+                .onErrorResume(closeError -> Completable.failed(addSuppressed(error, closeError)))
```

Comment:
nit: `closeError` could also be a `drainError`? maybe refer to it as `suppressedError` or similar since that's where it ends up?